micrometer-core/build.gradle

```diff
@@ -125,6 +125,9 @@ dependencies {
     optionalApi project(':micrometer-observation')
     optionalApi 'io.micrometer:context-propagation'
 
+    // Common Tests
+    testImplementation project(':micrometer-test')
```

Comment:
micrometer-test depends on micrometer-core. This is a circular dependency.

Reply:
Yeah, I just realised. I can remove this as this is not essential for this change.

Reply:
Since it is used in test scope only, I think we can use it, on further thought. I didn't look into the details of its use yet to give any thought to whether it makes sense or not, though.